mdpow/workflows/dihedrals.py

```diff
@@ -252,15 +281,29 @@ def dihedral_groups(dirname, resname, SMARTS=SMARTS_DEFAULT):
 
     '''
 
-    u = build_universe(dirname=dirname)
-    solute = rdkit_conversion(u=u, resname=resname)[1]
-    atom_group_indices = dihedral_indices(dirname=dirname, resname=resname, SMARTS=SMARTS)
-    dihedral_groups = [solute.atoms[list(dihedral_indices)].names for dihedral_indices
-                       in atom_group_indices]
+    dihedral_groups = [solute.atoms[list(a_ind)].names for a_ind
+                       in atom_indices]
 
     return dihedral_groups
 
-def dihedral_groups_ensemble(dirname, atom_group_indices,
+def get_paired_indices(atom_indices, bond_indices, dihedral_groups):
+
+    dg_list = []
+    for dg in dihedral_groups:
+        group = (f'{dg[0]}-{dg[1]}-{dg[2]}-{dg[3]}')
+        dg_list.append(group)
+    all_dgs = tuple(dg_list)
+
+    if len(atom_indices) == len(bond_indices) == len(all_dgs):
+        ab_pairs = {}
+        i = 0
+        while i < len(all_dgs):
+            ab_pairs[f'{all_dgs[i]}'] = (atom_indices[i], bond_indices[i])
+            i += 1
```

Comment:
Can't we write this with a for loop
```python
for i in range(len(all_dgs)):
   ab_pairs[f'{all_dgs[i]}'] = (atom_indices[i], bond_indices[i])
```
Much cleaner than while.

Reply:
Actually, even better do a [dict comprehension](https://docs.python.org/3/tutorial/datastructures.html#dictionaries)
```python
ab_pairs = {all_dgs[i]: (atom_indices[i], bond_indices[i]) for i in range(len(all_dgs))}
```
Please change to make the code more contact (and faster, too, even though that's not a major concern here)

Reply:
I did not know you could do that, but immediately realized how it works compared to the current method, very cool!!